### PR TITLE
Add the option to silence output from the X509 create_private_key function

### PR DESCRIPTION
**Pull request by akissa**
### Previous Behavior

Output was verbose with dots indicating the process of the key generation
### New Behavior

It is now possible to silence the verbose output
### Tests written?

Yes/No

progress status output to stdout, in some cases this
is not convenient as it mixes into the current stdout.

This is due to the underlying M2Crypto's modules
keygen_callback function which writes dots as
progress feedback to stdout

This change adds the ability to silence the output from
the function if you do now what it.

### CODE REVIEW

File: salt/modules/x509.py [hunk 1]
```diff
@@ -650,6 +658,9 @@ def create_private_key(path=None, text=False, bits=2048):
     bits:
         Length of the private key in bits. Default 2048
 
+    verbose:
+        Provide visual feedback on stdout. Default False
+
```
Comment (rallytime) [inline salt/modules/x509.py]: Can you add a `.. versionadded:: Carbon` tag here?

File: salt/modules/x509.py [hunk 2]
```diff
@@ -661,7 +672,12 @@ def create_private_key(path=None, text=False, bits=2048):
     if path and text:
         raise salt.exceptions.SaltInvocationError('Either path or text must be specified, not both.')
 
-    rsa = M2Crypto.RSA.gen_key(bits, M2Crypto.m2.RSA_F4)            # pylint: disable=no-member
+    if verbose:
+        _callback_func = M2Crypto.RSA.keygen_callback
+    else:
+        _callback_func = _keygen_callback
```
Comment (rallytime) [inline salt/modules/x509.py]: Is it necessary to use the _keygen_callback function here? Is it possible to just set this to `None`?

File: salt/states/x509.py [hunk 3]
```diff
@@ -222,6 +223,10 @@ def private_key_managed(name,
         When replacing an existing file, backup the old file onthe minion.
         Default is False.
 
+    verbose:
+        Provide visual feedback on stdout, dots while key is generated.
+        Default is False.
+
```
Comment (rallytime) [inline salt/states/x509.py]: A versionadded here, too, would be great!

**Comment by rallytime** [comment]
Hi @akissa - thank you for submitting this new option. I had a couple of comments/questions, but once those are addressed we should be able to get this in.

**Comment by akissa** [comment]
@rallytime Yes you need a callable None will not cut it based on the docs, have not checked the m2crypto source code though - https://www.heikkitoivonen.net/m2crypto/api/M2Crypto.RSA-module.html#gen_key

Will add versioning info

File: salt/modules/x509.py [hunk 4]
```diff
@@ -637,7 +645,7 @@ def write_pem(text, path, pem_type=None):
     return 'PEM written to {0}'.format(path)
 
 
-def create_private_key(path=None, text=False, bits=2048):
+def create_private_key(path=None, text=False, bits=2048, verbose=False):
```
Comment (rallytime) [inline salt/modules/x509.py]: I think we likely will want to preserve current functionality by setting versbose to True as a default. That way we won't surprise people during upgrades, but the option is there to turn it off.

File: salt/states/x509.py [hunk 5]
```diff
@@ -203,7 +203,8 @@ def _revoked_to_list(revs):
 def private_key_managed(name,
                         bits=2048,
                         new=False,
-                        backup=False):
+                        backup=False,
+                        verbose=False,):
```
Comment (rallytime) [inline salt/states/x509.py]: Same comment here about the default value. We should default this to True so we don't surprise people who may be relying on the current functionality.

**Comment by akissa** [comment]
@rallytime Updated.

**Comment by rallytime** [comment]
Awesome! Thanks @akissa! We'll let tests run on this and get it in.

**Pull request merged**